packages/cosmwasm-stargate/src/aminotypes.ts

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import {
+  MsgClearAdmin,
+  MsgExecuteContract,
+  MsgInstantiateContract,
+  MsgMigrateContract,
+  MsgStoreCode,
+  MsgUpdateAdmin,
+} from "@cosmjs/cosmwasm-launchpad";
+import { fromAscii, fromBase64, toAscii, toBase64 } from "@cosmjs/encoding";
+import { Coin } from "@cosmjs/launchpad";
+import { AminoConverter, codec } from "@cosmjs/stargate";
+import { assert } from "@cosmjs/utils";
+import Long from "long";
+
+import { cosmwasm } from "./codec";
+
+type ICoin = codec.cosmos.base.v1beta1.ICoin;
+type IMsgStoreCode = cosmwasm.wasm.v1beta1.IMsgStoreCode;
+type IMsgInstantiateContract = cosmwasm.wasm.v1beta1.IMsgInstantiateContract;
+type IMsgUpdateAdmin = cosmwasm.wasm.v1beta1.IMsgUpdateAdmin;
+type IMsgClearAdmin = cosmwasm.wasm.v1beta1.IMsgClearAdmin;
+type IMsgExecuteContract = cosmwasm.wasm.v1beta1.IMsgExecuteContract;
+type IMsgMigrateContract = cosmwasm.wasm.v1beta1.IMsgMigrateContract;
+
+function checkAmount(amount: readonly ICoin[] | undefined | null): readonly Coin[] {
+  assert(amount, "missing amount");
+  return amount.map((a) => {
+    assert(a.amount, "missing amount");
+    assert(a.denom, "missing denom");
+    return {
+      amount: a.amount,
+      denom: a.denom,
+    };
+  });
+}
+
+export const defaultTypes: Record<string, AminoConverter> = {
+  "/cosmwasm.wasm.v1beta1.MsgStoreCode": {
+    aminoType: "wasm/MsgStoreCode",
+    toAmino: ({ sender, wasmByteCode, source, builder }: IMsgStoreCode): MsgStoreCode["value"] => {
+      assert(sender, "missing sender");
+      assert(wasmByteCode, "missing wasmByteCode");
+      assert(typeof source === "string", "missing source");
+      assert(typeof builder === "string", "missing builder");
```

Comment:
It would be nice if `assertDefined` did that directly by changing it's signature to

```ts
function assertDefined<T>(value: T | undefined | null, msg?: string): asserts value is T;
```

and adding two checks internally. This would also improve other existing code like

```ts
export function coinFromProto(input: ICoin): Coin {
  assertDefined(input.amount);
  assertDefined(input.denom);
  assert(input.amount !== null);
  assert(input.denom !== null);
  return {
    amount: input.amount,
    denom: input.denom,
  };
}
```

The test

```
      {
        const value: null | undefined = null;
        assertDefined(value);
        expect(value).toEqual(null);
      }
```

should then be removed.